src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs

```diff
@@ -477,29 +481,44 @@ public DocumentConflict ResolveToLatest(List<DocumentConflict> conflicts)
         private void ResolveAttachmentsConflicts(DocumentsOperationContext context, DocumentConflict resolved, bool resolvedToLatest)
         {
             using var scope = Slice.External(context.Allocator, resolved.LowerId, out var lowerId);
-            var storageAttachmentsDetails = _database.DocumentsStorage.AttachmentsStorage.GetAttachmentDetailsForDocument(context, lowerId);
-            List<AttachmentName> resolvedAttachmentsMetadata = null;
+            var resolvedAttachmentsMetadata = AttachmentsStorage.GetAttachmentsFromDocumentMetadata(resolved.Doc)
```

Comment:
this can be moved down? 

since the purpose of that is to seek, then maybe it would be better to build a dict here

```
struct AttachmentMetadataKey
{
    string Name;
    string Hash;
    string ContentType;
}

Dictionary<AttachmentMetadataKey, AttachmentDetails>
```

instead of using `.Any` each time?

Reply:
The likely scenario is that you have, at max, a few dozens of attachments.
And conflicts are rare, I don't think it would be worth it